app/models/comment.rb

```diff
@@ -84,12 +85,24 @@ def no_grandparents
     errors.add(:parent, 'cannot have a parent of their own') if parent&.parent
   end
 
+  def ama_closed
+    return unless post
+    return unless post.ama
+    ama = post.ama
+    return if ama.author == user
+    now_time = Time.now
+
+    unless ama.start_date <= now_time && ama.end_date >= now_time
+      errors.add(:post, 'cannot make any more comments on this ama')
+    end
+  end
+
```

Comment:
Feeds can have a lot more than just posts/comments, we can send anything to a feed, just by mixing in `WithActivity` into the model.  I guess you'd also have to duplicate PostLikes/CommentLikes though, so it might get a little excessive to split it into its own model